src/pathfora.js

```diff
@@ -1296,77 +1439,105 @@
-          var element = getFormElement(field);
+            if (typeof child.getAttribute !== 'undefined') {
+              childName = child.getAttribute('name');
 
-          if (element && typeof element.placeholder !== 'undefined') {
-            element.placeholder = config.placeholders[field];
-          } else if (element && typeof element.options !== 'undefined') {
-            element.options[0].innerHTML = config.placeholders[field];
+              if (childName != null) {
+                form.removeChild(child);
+                k--;
+              }
+            }
           }
-        });
 
-        // Set required Fields
-        Object.keys(config.required).forEach(function (field) {
-          var element = getFormElement(field);
+          core.buildWidgetForm(config.formElements, form);
 
-          if (element && config.required[field]) {
-            element.setAttribute('data-required', 'true');
-          }
-        });
+        } else {
+          // suport old form functions
+          var getFormElement = function (field) {
+            if (field === 'name') {
+              return widget.querySelector('input[name="username"]');
+            }
 
-        // Hide fields
-        Object.keys(config.fields).forEach(function (field) {
-          var element = getFormElement(field),
-              parent = element.parentNode;
+            return widget.querySelector('form [name="' + field + '"]');
+          };
 
-          if (element && !config.fields[field] && parent) {
-            parent.removeChild(element);
-          }
-        });
+          // Set placeholders
+          Object.keys(config.placeholders).forEach(function (field) {
+            var element = getFormElement(field);
+
+            if (element && typeof element.placeholder !== 'undefined') {
+              element.placeholder = config.placeholders[field];
+            } else if (element && typeof element.options !== 'undefined') {
+              element.options[0].innerHTML = config.placeholders[field];
+            }
+          });
 
-        // NOTE: collapse half-width inputs
-        Array.prototype.slice.call(widget.querySelectorAll('form .pf-field-half-width')).forEach(function (element, halfcount) {
-          var parent = element.parentNode,
-              prev = element.previousElementSibling,
-              next = element.nextElementSibling;
+          // Set required Fields
+          Object.keys(config.required).forEach(function (field) {
+            var element = getFormElement(field);
 
-          if (parent) {
-            if (element.className.indexOf('pf-field-half-width') !== -1) {
+            if (element && config.required[field]) {
+              element.setAttribute('data-required', 'true');
+            }
+          });
+
+          // Hide fields
+          Object.keys(config.fields).forEach(function (field) {
+            var element = getFormElement(field),
+                parent = element.parentNode;
```

Comment:
this will throw an error if the element is not found (its happening in the existing legacy form example because the key is wrong `referrer` should be `referralEmail`. regardless, we probably want to gracefully handle the issue here by checking for a value and throwing an error if its null or whatever.